File: todolist/src/app/dto/TodoFilters.ts
import {Priority} from "../model/priority";

export interface TodoFilters {

  title: string | null;
  description: string | null;
  priority: Array<Priority> | null;
  startCreationTime: number | null;
  endCreationTime: number | null;
  startTime: number | null;
  endTime: number | null;
  scheduled: boolean | null;
  done: boolean | null;
  page: number | null;
  pageSize: number | null;
}

File: src/main/java/com/app/todo/dto/TodoFilters.java
package com.app.todo.dto;

import com.app.todo.model.Priority;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Collection;
import java.util.Date;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class TodoFilters {

    private String title;

    private String description;

    private Collection<Priority> priority;

    private Date startCreationDate;

    private Date endCreationDate;

    private Date startDate;

    private Date endDate;

    private boolean schedule;

    private Boolean done;

    private Integer page;

    private Integer pageSize;
}
